internal/executor/reporter/event_sender.go

```diff
@@ -0,0 +1,100 @@
+package reporter
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/armadaproject/armada/internal/common"
+	"github.com/armadaproject/armada/internal/common/eventutil"
+	"github.com/armadaproject/armada/pkg/api"
+	"github.com/armadaproject/armada/pkg/armadaevents"
+	"github.com/armadaproject/armada/pkg/executorapi"
+)
+
+type EventSender interface {
+	SendEvents(events []EventMessage) error
+}
+
+type ExecutorApiEventSender struct {
+	eventClient executorapi.ExecutorApiClient
+}
+
+func NewExecutorApiEventSender(executorApiClient executorapi.ExecutorApiClient) *ExecutorApiEventSender {
+	return &ExecutorApiEventSender{
+		eventClient: executorApiClient,
+	}
+}
+
+func (eventSender *ExecutorApiEventSender) SendEvents(events []EventMessage) error {
+	sequences := make([]*armadaevents.EventSequence, 0, len(events))
+	for _, e := range events {
+		message, err := api.Wrap(e.Event)
+		if err != nil {
+			return err
+		}
+		log.Debugf("Reporting event %+v", message)
+		sequence, err := eventutil.EventSequenceFromApiEvent(message)
+		if err != nil {
+			return err
+		}
+		runId, err := armadaevents.ProtoUuidFromUuidString(e.JobRunId)
+		if err != nil {
+			return fmt.Errorf("Failed to convert uuid string %s to uuid because %s", e.JobRunId, err)
+		}
+		populateRunId(sequence, runId)
+		sequences = append(sequences, sequence)
+	}
+	sequences = eventutil.CompactEventSequences(sequences)
```

Comment:
You might want to limit the size here else you could end up wit a mega event! I think there's something in eventutil to do this.